### PR TITLE
Fixed the return type of parse function of ConjunctiveGraph (Issue #939)

### PR DESCRIPTION
**Pull request by arnavpisces**
Instead of creating a new graph and parsing changes in that (therefore changing the graph type), the parsing is done in the graph itself and a self is returned. (Similar code as parse function of class Graph).

### CODE REVIEW

**Comment by nicholascar** [comment]
Hi @arnavpisces. Looks good overall but please can you supply tests? We really can't accept PRs to the core `graph.py` module without tests. Thanks.

**Comment by aucampia** [comment]
tagging related issue #939

**Comment by nicholascar** [comment]
Close/open to trigger Drone

**Comment by ashleysommer** [comment]
Hi @arnavpisces are you able to rebase this PR on latest master branch? We need the Drone configs in place on your branch in order to re-run the tests.

File: rdflib/graph.py [hunk 1]
```diff
@@ -1508,6 +1510,9 @@ def context_id(self, uri, context_id=None):
             context_id = "#context"
         return URIRef(context_id, base=uri)
 
+    # def identifier(self,id):
+    #     self._identifier=id
+
```
Comment (ashleysommer) [inline rdflib/graph.py:1515]: Remove these commented lines

File: rdflib/graph.py [hunk 2]
```diff
@@ -1538,15 +1543,29 @@ def parse(
             data=data,
             format=format,
         )
-
+        
+        # print(source.getPublicId(),publicID)
+
```
Comment (ashleysommer) [inline rdflib/graph.py:1548]: Remove these commented lines

File: rdflib/graph.py [hunk 3]
```diff
@@ -1538,15 +1543,29 @@ def parse(
...
             g_id = URIRef(g_id)
+            
+        super(ConjunctiveGraph, self).__init__(store=self.store,identifier=g_id)
+        super(ConjunctiveGraph,self).remove((None,None,None))
+        if format is None:
+            format = source.content_type
+        if format is None:
+            # raise Exception("Could not determine format for %r. You can" + \
+            # "expicitly specify one with the format argument." % source)
+            format = "application/rdf+xml"
```
Comment (ashleysommer) [inline rdflib/graph.py:1560]: Why is there an exception commented out here?

**Review by ashleysommer — CHANGES_REQUESTED** [review]

**Comment by gjhiggins** [comment]
Timed out.

**Pull request closed without merging**